packages/toolpad-core/src/client/layout/DashboardLayout/DashboardLayout.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+import {
+  AppBar,
+  Box,
+  Collapse,
+  Container,
+  Divider,
+  Drawer,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  ListSubheader,
+  Stack,
+  Toolbar,
+  Typography,
+} from '@mui/material';
+// Must use named imports or app crashes when the component is imported in a Next.js app
+// eslint-disable-next-line no-restricted-imports
+import { ExpandLess as ExpandLessIcon, ExpandMore as ExpandMoreIcon } from '@mui/icons-material';
+import { styled } from '@mui/system';
+import Link from 'next/link';
```

Comment:
No coupling with any router

```suggestion
```

Reply:
@apedroferreira perhaps to get a baseline version merged we can start with plain `a` tags? Then progressively improve with a pluggable router adapter and/or Link component?

Reply:
What I've been prototyping with is an [adapter](https://github.com/Janpot/dashboard-framework-prototype/blob/412060d835deffc3bb348e2f77bbf7ae1291d945/src/lib/dash/filter.tsx#L134-L188) that is modeled as a subset of the new [navigation API](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API). That way hopefully we can some day just delegate directly to `window.navigation` as the default router adapter. It's a challenge though to bridge to this from the different Next.js routers.

Reply:
I'm also using `@mui/material-nextjs/v14-appRouter` but the pages router should use `'@mui/material-nextjs/v14-pagesRouter'` and Vite shouldn't need anything... So I guess we need to allow the user to choose a framework somehow and have a component that uses one of these according to that setting.

Reply:
that is what the top level provider is for

Reply:
I made this version framework agnostic for now but if I use `<a>` tags I need to use something like `window.location.pathname` to compare routes, and this shows SSR errors in Next.js.
I can work on some kind of framework-aware navigation adapter in this PR already to solve this issue, or we can merge this and I can work on it next.